app/models/vm_migrate_workflow.rb

```diff
@@ -16,7 +16,8 @@ def self.default_dialog_file
   def get_source_and_targets(refresh = false)
     return @target_resource if @target_resource && refresh == false
 
-    ems = @values[:src_ids].to_miq_a.collect { |v_id| v = Vm.find_by_id(v_id); v.ext_management_system }.uniq.compact
+    vms = @values[:src_ids].to_miq_a.collect { |v_id| Vm.find_by_id(v_id) }
+    ems = vms.map(&:ext_management_system).uniq.compact
```

Comment:
I think its more effective from the point of view of database calls and memory to do:
ems_ids = Vm.where(:id => @values[:src_ids]).distinct.pluck(:ems_id)
#can there be severat ems? cause then its emses or something....
emses = ExtManagementSystem.where(:id => ems_ids)